src/main/java/seedu/address/model/person/Transaction.java

```diff
@@ -11,26 +11,36 @@
  */
 public class Transaction {
 
+    public static final String MESSAGE_CONSTRAINTS =
+            "Amount should have at most 2 decimal places.";
+    public static final String VALIDATION_REGEX = "/^-?\\d+(,\\d{3})*(\\.\\d{1,2})?$/";
     private final String description;
-    private final int amount;
+    private final double amount;
     private final String otherParty;
     private final LocalDate date;
 
     /**
      * Every field must be present and not null.
      */
-    public Transaction(String description, int amount, String otherParty, LocalDate date) {
+    public Transaction(String description, double amount, String otherParty, LocalDate date) {
         requireAllNonNull(description, amount, otherParty, date);
         this.description = description;
         this.amount = amount;
         this.otherParty = otherParty;
         this.date = date;
     }
 
+    /**
+     * Returns true if a given string is a valid phone number.
+     */
+    public static boolean isValidTransaction(String test) {
```

Comment:
Can change the method header and the method name to isValidAmount, I think can merge after that

Reply:
Ok, thank you!